File: 20.cpp
#include <iostream>
#include <list>
#include <vector>
using namespace std;

// Issue : object class, main class, main function
class Test {
  public:
    static void F() { cout << ("F()") << endl; }
//    static void F(object x) { cout << ("F(object)") << endl; }
    static void F(int x) { cout << ("F(int)") << endl; }
    static void F(double x) { cout << ("F(double)") << endl; }
    template<typename T> static void F(T x) { cout << ("F<T>(T)") << endl; }
    static void F(double x, double y) { cout << ("F(double, double)") << endl; }
};

int main(){
  Test::F();
  Test::F(1);
  Test::F(1.0); 
  Test::F("abc"); 
  Test::F((double)1); 
//  Test::F((object)1); 
  Test::F<int>(1); 
  Test::F(1, 1);
}

File: 20.cs
using System;
using System.Collections.Generic;

// overloading 
class Test {
  static void F() { Console.WriteLine("F()"); }
  static void F(object x) { Console.WriteLine("F(object)"); }
  static void F(int x) { Console.WriteLine("F(int)"); }
  static void F(double x) { Console.WriteLine("F(double)"); }
  static void F<T>(T x) { Console.WriteLine("F<T>(T)"); }
  static void F(double x, double y) { Console.WriteLine("F(double, double)"); }
  static void Main() { 
    F();
    F(1);
    F(1.0); 
    F("abc"); 
    F((double)1); 
    F((object)1); 
    F<int>(1); 
    F(1, 1);
  }
}
